source/common/filesystem/posix/filesystem_impl.h

```diff
@@ -46,5 +46,6 @@ class InstanceImplPosix : public Instance {
   friend class FileSystemImplTest;
 };
 
+InstancePtr makeFilesystemInstance() { return std::make_unique<InstanceImplPosix>(); }
```

Comment:
Changes look good.  Removing #ifdefs certainly makes code easier to write.  What patterns / prior art exists for ways in which we create platform specific objects?

I wonder if we should consider something like:
`using InstanceImpl = InstanceImplPosix`

and
```
Envoy::Filesystem::InstanceImpl fs;
fs.doStuff(a, b, c);
```

Reply:
`using InstanceImpl ` is a neat idea. I changed it to that.

Reply:
It makes more sense since we are abstracting away a type